eventsource-client/src/client.rs

```diff
@@ -230,28 +248,51 @@ pub struct ReconnectingRequest<C> {
     #[pin]
     state: State,
     next_reconnect_delay: Duration,
+    event_parser: EventParser,
+    last_event_id: String,
 }
 
 impl<C> ReconnectingRequest<C> {
-    fn new(http: hyper::Client<C>, props: RequestProps) -> ReconnectingRequest<C> {
+    fn new(
+        http: hyper::Client<C>,
+        props: RequestProps,
+        last_event_id: String,
+    ) -> ReconnectingRequest<C> {
         let reconnect_delay = props.reconnect_opts.delay;
         ReconnectingRequest {
             props,
             http,
             state: State::New,
             next_reconnect_delay: reconnect_delay,
+            event_parser: EventParser::new(),
+            last_event_id,
         }
     }
 
     fn send_request(&self) -> Result<ResponseFuture>
     where
         C: Connect + Clone + Send + Sync + 'static,
     {
-        let mut request = Request::get(&self.props.url);
-        *request.headers_mut().unwrap() = self.props.headers.clone();
-        let request = request
-            .body(Body::empty())
-            .map_err(|e| Error::InvalidParameter(Box::new(e)))?;
+        let mut request_builder = Request::builder().uri(&self.props.url);
+
+        for (name, value) in &self.props.headers {
+            request_builder = request_builder.header(name, value);
+        }
+
+        if !self.last_event_id.is_empty() {
+            request_builder = request_builder.header(
+                "last-event-id",
+                HeaderValue::from_str(&self.last_event_id.clone()).unwrap(),
+            );
+        }
+
+        let body = Body::empty();
+
+        // todo(cwaldren): Should this be BAD_REQUEST?
```

Comment:
Under what situations will body fail?

Reply:
I believe this should be `Error::InvalidParameter(Box::new(e))`, after reviewing my previous mixup with BAD_REQUEST.

> This function may return an error if any previously configured argument failed to parse or get converted to the internal representation. For example if an invalid head was specified via header("Foo", "Bar\r\n") the error will be returned when this function is called rather than when header was called.